### DISCUSSION TITLE
[FEATURE] Switch from Create React App to Vite?

### DISCUSSION THREAD

**Discussion by dan-mba** [EddieHubCommunity/EddieHubLive#69] [Ideas]
### Description

I noticed this project is using Create React App and was wondering if you would be interested in switching to [Vite](https://vitejs.dev/).

Vite has faster builds and is more configurable.

**Comment by amandamartin-dev** [comment]
Hey @dan-mba sorry for the delay responding to this.  I think this would be best opened up as a discussion.  I've read about Vite but haven't actually used it yet.  Would you like to start the convo in the discussions area and other folks can weigh in as well?

**Comment by dan-mba** [comment]
> Hey @dan-mba sorry for the delay responding to this.  I think this would be best opened up as a discussion.  I've read about Vite but haven't actually used it yet.  Would you like to start the convo in the discussions area and other folks can weigh in as well?

Can you convert this issue to a discussion?

It requires write access to the repo.
Here are the [instructions](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion)

**Reply by amandamartin-dev** [reply]
Thanks! Completely missed that.  

So I've read a bit about Vite and see how it's considered a better option for production in many cases but curious if it is necessary for a smaller project like this that has a very specific use case.  Curious what you are thinking is the benefit here?  Also curious what others are thinking as well.

**Comment by dan-mba** [comment]
The main benefit of moving to Vite is improved speed in development & production build.
An explanation of how Vite does this is on [their website(https://vitejs.dev/guide/why.html).

Vite is also updated regularly, so it reduces the need for `package.json` overrides or eject workarounds.

Vite is used by some newer frameworks like Astro & SvelteKit.
I personally use it on all of my React projects that don't use a framework like Next or Gatsby.

**Reply by Cahllagerfeld** [reply]
I'd also be in favor of migrating for the given reasons!

**Reply by amandamartin-dev** [reply]
Thank you so much for weighing in! I agree with Dan too just haven't used Vite yet so appreciate your thoughts as well.

**Reply by dan-mba** [reply]
Ok, I will work on a PR for this migration.

**Comment by Cahllagerfeld** [comment]
I was hacking around with it, see on the [new branch](https://github.com/EddieHubCommunity/EddieHubLive/tree/cahllagerfeld/migrate-to-vite). I didn't open the PR just yet though, because it's not properly tested (yet)

**Reply by amandamartin-dev** [reply]
You are so fast! I will take a look later and play around. @dan-mba would love your thoughts on the branch too as you have much experience with Vite

**Reply by dan-mba** [reply]
A couple of thoughts:

You can remove all of the `@testing-library` dependencies as you aren't using them (CRA added them by default)
I would recommend installing `eslint` & `eslint-plugin-react` (linting was setup by default with CRA)
Migrate eslint config out of package.json `npx eslint --init` (The package.json setup doesn't work well with vscode).
Try removing the overrides section of `package.json` (from what I can tell, it is not a sub-dependency of anything else)

If you need any help, at me on GitHub or mention me on EddieHub discord ( my id is dan-mba there too)
